src/WebJobs.Script.WebHost/Diagnostics/WebJobsEventListener.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Diagnostics
+{
+    internal class WebJobsEventListener : EventListener
+    {
+        private readonly ILogger _logger;
+
+        public WebJobsEventListener(WebHostSystemLoggerProvider systemLoggerProvider)
+        {
+            _logger = systemLoggerProvider.CreateLogger(GetType().FullName);
+        }
+
+        protected override void OnEventSourceCreated(EventSource eventSource)
+        {
+            if (eventSource.Name.StartsWith("Microsoft-ApplicationInsights"))
+            {
+                EnableEvents(eventSource, EventLevel.Error);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            switch (eventData.Level)
+            {
+                case EventLevel.Critical:
+                case EventLevel.Error:
+                    if (TryFormatMessage(eventData, out string message))
+                    {
+                        _logger.LogError(message);
```

Comment:
maybe this is duplicated same logError from inside TryFormatMessage

Reply:
The only time we'll log the error inside TryFormatMessage is if the formatting failed, which means TryFormatMessage will return false, and we'll skip this log. So it will always be one or the other.

I'm being paranoid with the formatting exception handling -- but I don't want to accidentally throw in this code and cause some weird, untrackable crash.

Reply:
i misread the code, ignore my comment :)